modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -153,45 +169,80 @@ SynonymMap buildSynonyms(Analyzer analyzer, ReaderWithOrigin rules) {
     }
 
     protected ReaderWithOrigin getRulesFromSettings(Environment env, IndexCreationContext context) {
-        if (settings.getAsList("synonyms", null) != null) {
-            List<String> rulesList = Analysis.getWordList(env, settings, "synonyms");
-            StringBuilder sb = new StringBuilder();
-            for (String line : rulesList) {
-                sb.append(line).append(System.lineSeparator());
+        return switch (synonymsSource) {
```

Comment:
I like the structure using enum. Should we take that a level further, and make the enums have a method to return the ReaderWithOrigin so we avoid the switch here?